crypto/src/merkle_tree/merkle_tree.rs

```diff
@@ -0,0 +1,241 @@
+use std::{cell::RefCell, rc::Rc};
+use crate::hash::traits::IsCryptoHash;
+use lambdaworks_math::field::{element::FieldElement, traits::IsField};
+
+pub struct MerkleTree<F: IsField, H: IsCryptoHash<F>> {
+    leafs: Vec<TreeNode<F>>,
+    root: TreeNode<F>,
+    hasher: H,
+}
+
+impl<F: IsField, H: IsCryptoHash<F>> MerkleTree<F, H> {
+    pub fn build(values: &[FieldElement<F>], hasher: H) -> MerkleTree<F, H> {
+        let mut leafs: Vec<TreeNode<F>> = hash_leafs(values, &hasher);
+        let mut level: Vec<TreeNode<F>> = hash_level(&mut leafs, &hasher);
+
+        while level.len() > 1 {
+            level = hash_level(&mut level, &hasher);
+        }
+
+        return MerkleTree {
```

Comment:
This `return` is not needed.

Reply:
yes fixed those kind of warnings in the following commit